### DISCUSSION TITLE
Limit number of cli.wrap running at a time

### DISCUSSION THREAD

**Discussion by MikeAtTRP** [Tyrrrz/CliWrap#250] [Q&A] [answered]
Hi I have clip.wrap running a large number of ffmpeg encode processes (50-100) every 5 mins at the moment they all start at once and run the cpu at 100% , i would like to set a limit on the number of commands executing at the same time.  MaxDegreeOfParallelism in Parallel.Foreach has no effect. is this possible  with SemaphoreSlim or something else

**Comment by Tyrrrz** [comment, marked as answer]
Yes, use a semaphore slim, blocking before `ExecuteAsync()`.

**Reply by MikeAtTRP** [reply]
Thank you that has done the trick

Mike
